zulip_bots/zulip_bots/bots/lunch/lunch.py

```diff
@@ -0,0 +1,219 @@
+
+                <user|optional> order
+                Shows all items of the sender's (or <user>'s if specified) list of items
+
+                who should <verb>(?)
+                Randomly selects a user from the orders list
+
+                help
+                Shows help (this section)
+            '''
+
+        return "I don't think I can do that. :frowning:"
+
+# User finder
+# Finds if a user exists
+# error is True, return an error message
+# error is False, create the user line
+def FIND_USER(user, error):
+    try:
+        ORDERS[user]
+    except KeyError:
+        if error:
+            return 'Sry, @**' + user + "** isn't hungry. :disappointed:"
+        else:
+            ORDERS[user] = []
+            return
+
+# Item interpreter
+# Transforms an array of items into a string
+def ITEM_INTERPRETER(items):
+    item_str = ''
+    for item in items:
+        if len(item_str) == 0:
+            item_str += ' ' + item
+        else:
+            if item_str[-1] == ',':
+                item_str += item
+            else:
+                item_str += ' ' + item
+
+    return item_str[1:].split(',')
+
+# Copies user list
+def copy_user_list(original_user, user):
+    # Verification
+    if FIND_USER(original_user, True):
+        return FIND_USER(original_user, True)
+    FIND_USER(user, False)
+
+    # Copies list
+    ORDERS[user].extend(ORDERS[original_user])
+
+    return "**Here's the new** @**" + user + '** **list of items:** :wink:\n' + ', '.join(ORDERS[user])
+
+# Adds an item to a list
+def add_to_user_list(user, items):
+    # Verification
+    FIND_USER(user, False)
+
+    # Add items
+    items = ITEM_INTERPRETER(items)
+    ORDERS[user].extend(items)
+
+    return "**Here's the new** @**" + user + '** **list of items:** :wink:\n' + ', '.join(ORDERS[user])
+
+# Removes an item from a list
+def remove_from_user_list(user, items):
+    # Verification
+    if FIND_USER(user, True):
+        return FIND_USER(user, True)
+
+    # Removes items
+    items = ' '.join(items)
+    # Verify if it's an array fo items or not
+    if ',' in items:
+        items = items.split(', ')
+        for item in items:
+            try:
+                ORDERS[user].remove(item)
+            except ValueError:
+                return "@**" + user + "** ** doesn't have " + item + " in the list** :neutral_face:"
+    else:
+        try:
+            ORDERS[user].remove(items)
+        except ValueError:
+            return "@**" + user + "** ** doesn't have " + items + " in the list** :neutral_face:"
+
+    return "**Here's the new** @**" + user + '** **list of items:** :wink:\n' + ', '.join(ORDERS[user])
+
+# Removes the list of items
+def remove_all(user):
+    # Verification
+    if FIND_USER(user, True):
+        return FIND_USER(user, True)
+
+    # Removes user
+    del ORDERS[user]
+
+    return 'Cleared @**' + user + '** list! :ok_hand:'
+
+# Shows all orders
+def show_all_orders():
+    order = str(ORDERS)
```

Comment:
This feels like a strange way to build the list of orders.  You can do something more like `', '.join(ORDERS)` to produce the list.